src/easylink/rule.py

```diff
@@ -46,8 +49,13 @@ def _build_rule(self) -> str:
         term="spark_logs/spark_master_terminated.txt",
         master_log="spark_logs/spark_master_log.txt",
         worker_logs=gather.num_workers("spark_logs/spark_worker_log_{{scatteritem}}.txt",
-        ),
-            """
+        ),"""
+        rulestring += f"""
+    output: {outputs}
+    run:
+        import os
+        for input_path, output_path in zip(input.final_output, output):
```

Comment:
symlink whatever the proper set of intermediate results (in this case just the output of step 4) to the base directory. This has the potential for collisions, but that might not end up mattering if we expect to only ever get one big final output file.